anta/tests/routing/bgp.py

```diff
@@ -746,20 +800,14 @@ def test(self) -> None:
                 failures = deep_update(failures, failure)
                 continue
 
-            # Check each capability
+            # Fetching the capabilities output.
             bgp_output = get_value(bgp_output, "neighborCapabilities.multiprotocolCaps")
-            for capability in capabilities:
-                capability_output = bgp_output.get(capability)
-
-                # Check if capabilities are missing
-                if not capability_output:
-                    failure["bgp_peers"][peer][vrf][capability] = "not found"
-                    failures = deep_update(failures, failure)
-
-                # Check if capabilities are not advertised, received, or enabled
-                elif not all(capability_output.get(prop, False) for prop in ["advertised", "received", "enabled"]):
-                    failure["bgp_peers"][peer][vrf][capability] = capability_output
-                    failures = deep_update(failures, failure)
+
+            # Collecting the failure logs if any.
+            failure_log = _check_peer_capabilities(bgp_output, capabilities, strict=bgp_peer.strict)
+            if failure_log:
+                failure["bgp_peers"][peer][vrf] = failure_log
+                failures = deep_update(failures, failure)
```

Comment:
I am not entirely sure if we _should_ use `deep_update` since it's from Pydantic v1. Can you investigate if we can use something else or have a util function doing the same?

Reply:
As It is existing test case, kept this deep update as it is.